mealy/error_visualizer.py

```diff
@@ -199,55 +225,54 @@ def plot_feature_distributions_on_leaves(self, leaf_selector='all_errors',
+                * 'purity': rank by the purity (ratio of wrongly predicted samples over the total number of node samples)
+                * 'class_difference': rank by the difference of number of wrongly and correctly predicted samples
+                in a node.
 
             nr_bins (int): number of bins in the feature distribution plots.
 
             figsize (tuple): size of the plots.
 
         """
 
-        error_class_idx = np.where(self._error_clf.classes_ == ErrorAnalyzerConstants.WRONG_PREDICTION)[0][0]
-        correct_class_idx = 1 - error_class_idx
+        correct_class_idx = 1 - self._error_tree.error_class_idx
 
         ranked_feature_ids = rank_features_by_error_correlation(self._error_clf.feature_importances_)
-        if self.pipeline_preprocessor is None:
+        if self._error_analyzer.pipeline_preprocessor is None:
             if top_k_features > 0:
                 ranked_feature_ids = ranked_feature_ids[:top_k_features]
 
-            x, y = self._error_train_x[:, ranked_feature_ids], self._error_train_y
+            x, y = self._error_analyzer._error_train_x[:, ranked_feature_ids], self._error_analyzer._error_train_y
             min_values, max_values = x.min(axis=0), x.max(axis=0)
-            feature_names = self.mpp_feature_names
+            feature_names = self._error_analyzer.preprocessed_feature_names
         else:
-            ranked_feature_ids = [self.pipeline_preprocessor.inverse_transform_feature_id(idx) for idx in
-                                  ranked_feature_ids]
+            ranked_feature_ids = list(set(self._error_analyzer.pipeline_preprocessor.inverse_transform_feature_id(idx) for idx in
+                                          ranked_feature_ids))
             if top_k_features > 0:
                 ranked_feature_ids = ranked_feature_ids[:top_k_features]
-            x, y = self.pipeline_preprocessor.inverse_transform(self._error_train_x)[:,
-                   ranked_feature_ids], self._error_train_y
-            # TODO
+
+            x, y = self._error_analyzer.pipeline_preprocessor.inverse_transform(self._error_analyzer._error_train_x)[:, ranked_feature_ids], self._error_analyzer._error_train_y
+            # TODO to do what ?
             min_values, max_values = x.min(axis=0), x.max(axis=0)
-            feature_names = self.original_feature_names
+            feature_names = self._original_feature_names
 
-        global_error_sample_ids = y == ErrorAnalyzerConstants.WRONG_PREDICTION
-        nr_wrong, nr_correct = self._error_clf.tree_.value[:, 0, error_class_idx], self._error_clf.tree_.value[:, 0,
-                                                                                   correct_class_idx]
+        total_error_fraction_sample_ids = y == ErrorAnalyzerConstants.WRONG_PREDICTION
+        nr_wrong, nr_correct = self._error_clf.tree_.value[:, 0, self._error_tree.error_class_idx], self._error_clf.tree_.value[:, 0, correct_class_idx]
```

Comment:
```suggestion
        nr_wrong = self._error_clf.tree_.value[:, 0, self._error_tree.error_class_idx]
```

- Also below (l.266-298):
```python
     leaf_sample_ids = self._train_leaf_ids == leaf
     nr_leaf_samples = self._error_clf.tree_.n_node_samples[leaf]
     proba_wrong_leaf = nr_wrong[leaf] / nr_leaf_samples
     proba_correct_leaf = 1 - proba_wrong_leaf
```

- Also l.307:
```python
    root_prediction = ErrorAnalyzerConstants.CORRECT_PREDICTION if nr_wrong[0] < self._error_clf.tree_.n_node_samples[0,0,self._error_tree.error_class_idx]/2 else ErrorAnalyzerConstants.WRONG_PREDICTION
```

Reply:
Done in 62bec2c